app/_src/gateway/upgrade/backup-and-restore.md

```diff
@@ -0,0 +1,270 @@
+---
+title: Backup and restore
+content_type: how-to
+purpose: Learn how to back up and restore Kong Gateway
+---
+
+Before you start any upgrade, back up your {{site.base_gateway}} data. 
+Kong supports two back up methods for {{site.base_gateway}} entities: [database-native backup](#database-native-backup) and [declarative backup](#declarative-backup).
+A database-native backup backs up the entire {{site.base_gateway}} database, while a declarative backup works by managing declarative configuration files.
+
+We recommend backing up data using both methods, as this offers recovery flexibility:
+* The database-native tools are robust and can restore data instantly, compared to the declarative tools. 
+* In case of data corruption, try to do a database-level restore first, otherwise bootstrap a new database and [declarative tools](#declarative-tools-for-backup-and-restore) to load in entity data.
+
+[Keyring materials](#keyring-materials-backup-and-restore) and {{site.base_gateway}} [configuration files](#other-files) must be backed up separately. 
+See their respective sections below for details.
+
+The backup and restore methods described in this guide serve as general instructions.
+Revise the methods as necessary to fit your infrastructure, deployment, and business requirements.
+
+## Declarative tools for backup and restore
+
+Kong ships two declarative backup tools: [decK](/deck/) and the [Kong CLI](/gateway/{{page.kong_version}}/reference/cli/), which support managing {{site.base_gateway}} entities in the declarative format. 
+
+* For **database-backed deployments** (traditional and hybrid mode), backups taken with either of these tools serve as an extra safeguard layer. If the database-native backup or restore corrupts the database, you can fall back to declarative files for restoring data.
+
+    Both tools require the database to be ready for data export and import. To import or export data using these tools, ensure the user and password are initialized, and the database is bootstrapped.
+
+* For **DB-less deployments**, no special tools are needed, so there is no declarative tool support. Back up your declarative files manually.
+
+decK is generally more powerful than the Kong CLI. It has more features, invalidates the cache automatically, and fetches entities from the database instead of the LRU cache. Additionally, it overwrites entities instead of patching, so that the database has the exact copy of the config that you provide.
```

Comment:
```suggestion
decK is generally more powerful than the `kong config` CLI. It has more features, invalidates the cache automatically, and fetches entities from the database instead of the LRU cache. Additionally, it overwrites entities instead of patching, so that the database has the exact copy of the config that you provide.
```